### DISCUSSION TITLE
How does Renovate find out duplicate PRs?

### DISCUSSION THREAD

**Discussion by rvarun11** [renovatebot/renovate#23594] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I wanted to know how Renovate decides whether the dependency upgrade it found is duplicate and sends the `Ignoring Upgrade Collision` log message. 

I created by custom maven manager which points to our internal datasource for fetching dependencies upgrades instead of maven datasource. 

When I run RenovateBot, I want the GCRs to be created by both original maven manager as well as the custom maven manager I made so that we can compare and constrast between the 3PPs generated by them and decide which one to keep.

### What you tried so far.

N/A

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
It's best that you search the source code for `Ignoring Upgrade Collision` to see it exactly. Was there anything there you couldn't understand?

**Comment by rvarun11** [comment, marked as answer]
Thanks for your response. The issue turned out to be because the branches created by both managers were same. So changing the `branchPrefix` for our custom manager fixed the issue.
